Templates/Templates/ModelMappable.stencil

```diff
@@ -0,0 +1,162 @@
+// https://github.com/SwiftGen/StencilSwiftKit/blob/master/Documentation/tag-set.md
+// https://github.com/kylef/Stencil
+// https://cdn.rawgit.com/krzysztofzablocki/Sourcery/master/docs/index.html
+
+import Foundation
+
+private typealias Timestamp = UInt64
+
+private struct Formatter {
+  static let iso8601: DateFormatter = {
+    let formatter = DateFormatter()
+    formatter.calendar = Calendar(identifier: .iso8601)
+    formatter.locale = Locale(identifier: "en_US_POSIX")
+    formatter.timeZone = TimeZone(abbreviation: "UTC")
+    formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSXXXXX"
+    return formatter
+  }()
+}
+
+extension String {
+  fileprivate var dateFromISO8601: Date? {
+    // "Mar 22, 2017, 10:22 AM"
+    return Formatter.iso8601.date(from: self)
+  }
+}
+
+extension Date {
+  fileprivate var iso8601: String {
+    return Formatter.iso8601.string(from: self)
+  }
+
+  fileprivate static func fromTimestamp(_ timestamp: Timestamp) -> Date {
+    return Date(timeIntervalSince1970: Double(timestamp) / 1000)
+  }
+
+  fileprivate var asTimestamp: Timestamp {
+    return UInt64(1000 * timeIntervalSince1970)
+  }
+}
+
+{% macro modelName name %}{{ name | replace:"Mapping","Model" }}{% endmacro %}
+{% macro remoteName var %}{% if var.annotations.coding %}{{ var.annotations.coding }}{% elif var.typeName.name == "String" %}{{ var.name|camelToSnakeCase|replace:"id","uuid" }}{% else %}{{ var.name|camelToSnakeCase }}{% endif %}{% endmacro %}
+
+{% macro immutableVariables variables %}
+  {% for var in variables %}
+  let {{ var.name }}: {{ var.typeName }}
+  {% endfor %}
+{% endmacro %}
+
+{% macro codingKeys variables %}
+  private enum CodingKeys: String, CodingKey {
+    {% for var in variables %}
+    case {{ var.name }} = "{% call remoteName var %}"
+    {% endfor %}
+  }
+{% endmacro %}
+
+{% macro decodeVar var %}
+  {% set itype %}{{ var.annotations.incomingType }}{% endset %}
```

Comment:
It might be that some api methods return the same models with different primitive types, strings in place of numbers and so on. Using annotation to fix type that comes in limits it to one particular scheme. I believe this should be handled by extensions on decoder instead that will not just decode bool from bool, but will support any compatible value (0/1, "true"/"false"), same for other types.

Reply:
Won't the expected type still need to be known for encoding the value correctly?

Reply:
After implementing it myself I actually found it useful to sometimes declare expected type of value, but that always comes with subsequent mapping of this type to the type of corresponding variable, i.e. String -> Double. So I'm using `decoding_type` and `decoding_map` annotations for that.